core/services/gateway/handlers/functions/subscriptions.go

```diff
@@ -0,0 +1,200 @@
+package functions
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+
+	evmclient "github.com/smartcontractkit/chainlink/v2/core/chains/evm/client"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/functions/generated/functions_router"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/job"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+type OnchainSubscriptionsConfig struct {
+	RouterAddress      common.Address `json:"routerAddress"`
+	BlockConfirmations uint           `json:"blockConfirmations"`
+	QueryFrequencySec  uint           `json:"queryFrequencySec"`
```

Comment:
To follow allowlist config fields, I'd suggest "UpdateFrequencySec", "UpdateTimeoutSec" and "UpdateRangeSec".